### PR TITLE
Update to allow custom HTML elements definitions

### PR DESCRIPTION
**Pull request by codel**
Just added an special option "definitions" in the config which allows to set custom HTML definitions, in order to add support of elements and attributes, like: <figure> or [allowfullscreen] for an iframe.

### CODE REVIEW

File: src/Purifier.php [hunk 1]
```diff
@@ -122,13 +122,51 @@ protected function getConfig($config = null)
             if( ! $this->config->get('purifier.settings.default'))
```
Comment (itbdw) [inline src/Purifier.php]: This is buggy when the settings does not have a definitions

File: src/Purifier.php [hunk 2]
```diff
@@ -122,13 +122,51 @@ protected function getConfig($config = null)
...
             {
             }
+            $chosen = 'purifier.settings.default.definitions';
```
Comment (itbdw) [inline src/Purifier.php]: This is buggy when the settings does not have a purifier.settings.default.definitions.

I think you should test enough before submit the pull request.

And to keep it simple, here should not place any value not compacted with params that needed to instance HTMLPurifier_Config.

@codel @mewebstudio

**Comment by hackel** [comment]
So...I was using this functionality to approximate HTML5 support in HTMLPurifier (taken from https://github.com/kennberg/php-htmlpurfier-html5).  We still need to find a way to implement this.

**Comment by azeos** [comment]
Any updates on this? How can I add HTML5 elements?

**Comment by itbdw** [comment]
@azeos I think it's better to use the official repo directly. See https://github.com/mewebstudio/Purifier/issues/32#issuecomment-225628250

**Comment by azeos** [comment]
@itbdw I'd already read your comment, but I'm not sure how to add the official package to laravel. I mean, in SO there is an example for Symfony -which I know is pretty similar-, but I don't know exactly what I have to do with Laravel.

May be this: https://laracasts.com/discuss/channels/tips/htmlpurifier-in-laravel-5/replies/20019 ?

Thanks!

**Comment by itbdw** [comment]
@azeos I have updated the comment. I think it's better that we can use the static method... 💯 https://github.com/mewebstudio/Purifier/issues/32#issuecomment-225628250

**Pull request merged**